src/main/java/com/efficio/fieldbook/web/common/controller/GermplasmTreeController.java

```diff
@@ -1017,155 +932,6 @@ protected boolean isSimilarToRootFolderName(final String itemName) {
-			gpList.setName(newName);
-
-			this.germplasmListManager.updateGermplasmList(gpList);
-
-			resultsMap.put(GermplasmTreeController.IS_SUCCESS, "1");
-		} catch (final Exception e) {
-			GermplasmTreeController.LOG.error(e.getMessage(), e);
-			resultsMap.put(GermplasmTreeController.IS_SUCCESS, "0");
-			resultsMap.put(GermplasmTreeController.MESSAGE, e.getMessage());
-		}
-		return resultsMap;
-	}
-
-	public boolean hasChildren(final Integer id, final String programUUID) {
-		return !this.germplasmListManager.getGermplasmListByParentFolderId(id, programUUID).isEmpty();
-	}
-
-	@ResponseBody
-	@RequestMapping(value = "/deleteGermplasmFolder", method = RequestMethod.POST)
-	public Map<String, Object> deleteGermplasmFolder(final HttpServletRequest req) {
-		final Map<String, Object> resultsMap = new HashMap<>();
-
-		final GermplasmList gpList;
-		final String folderId = req.getParameter("folderId");
-		try {
-			gpList = this.germplasmListManager.getGermplasmListById(Integer.parseInt(folderId));
-
-			if (this.hasChildren(gpList.getId(), this.getCurrentProgramUUID())) {
-				throw new MiddlewareException(GermplasmTreeController.HAS_CHILDREN);
-			}
-			this.germplasmListManager.deleteGermplasmList(gpList);
-			resultsMap.put(GermplasmTreeController.IS_SUCCESS, "1");
-		} catch (final Exception e) {
-			GermplasmTreeController.LOG.error(e.getMessage(), e);
-			resultsMap.put(GermplasmTreeController.IS_SUCCESS, "0");
-			resultsMap.put(GermplasmTreeController.MESSAGE, e.getMessage());
-		}
-		return resultsMap;
-	}
-
-	@ResponseBody
-	@RequestMapping(value = "/moveGermplasmFolder", method = RequestMethod.POST)
-	public Map<String, Object> moveStudyFolder(final HttpServletRequest req) {
-		final String sourceId = req.getParameter("sourceId");
-		final String targetId = req.getParameter("targetId");
-
-		final Map<String, Object> resultsMap = new HashMap<>();
-
-		try {
-
-			final GermplasmList gpList = this.germplasmListManager.getGermplasmListById(Integer.parseInt(sourceId));
-			GermplasmList parent = null;
-
-			if (targetId == null || PROGRAM_LISTS.equals(targetId) || CROP_LISTS.equals(targetId)) {
-				gpList.setParent(null);
-			} else {
-			    parent = this.germplasmListManager.getGermplasmListById(Integer.parseInt(targetId));
-				gpList.setParent(parent);
-			}
-
-			if (CROP_LISTS.equals(targetId) || (parent != null && StringUtils.isEmpty(parent.getProgramUUID()) && !StringUtils.isEmpty(
-				gpList.getProgramUUID()) && !GermplasmList.FOLDER_TYPE.equals(gpList.getType()))) {
-				gpList.setProgramUUID(null);
-				gpList.setStatus(LOCKED_LIST_STATUS);
-			} else {
-				gpList.setProgramUUID(this.contextUtil.getCurrentProgramUUID());
-			}
-
-			this.germplasmListManager.updateGermplasmList(gpList);
-
-		} catch (final Exception e) {
-			GermplasmTreeController.LOG.error(e.getMessage(), e);
-		}
-
-		return resultsMap;
-	}
-
 	@ResponseBody
 	@RequestMapping(value = "/save/state/{type}")
```

Comment:
Can we move saveTreeState and retrieveTreeState to StudyTreeController and update calls accordingly?